app/models/vm_or_template.rb

```diff
@@ -1881,10 +1881,16 @@ def self.includes_template?(ids)
 
   def self.cloneable?(ids)
     vms = VmOrTemplate.where(:id => ids)
-    return false if vms.blank?
+    return false if vms.blank? || vms.any? {|vm| vm.orphaned? || vm.archived? }
     vms.all?(&:cloneable?)
   end
 
+  def self.publishable?(ids)
+    vms = VmOrTemplate.where(:id => ids)
+    return false if vms.blank? || vms.any? {|vm| vm.orphaned? || vm.archived? }
+    vms.all?(&:publishable?)
```

Comment:
It's slightly better to call `all?` before `blank?`, if that's an easy rearrangement (which it should be, after addressing the above)